photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -245,13 +256,13 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
    * @return Normalization contexts for each featureShardId, or None if normalization is not needed
    */
   protected[training] def prepareNormalizationContexts(
-      trainingData: DataFrame,
+      trainingData: Option[DataFrame],
       featureIndexMapLoaders: IndexMapLoaders,
       statistics: FeatureShardStatisticsOpt): Option[Map[FeatureShardId, NormalizationContext]] =
 
     Filter(params.normalizationType != NormalizationType.NONE) {
       statistics
-        .getOrElse(calculateStatistics(trainingData, featureIndexMapLoaders))
+        .getOrElse(calculateStatistics(trainingData.get, featureIndexMapLoaders))
```

Comment:
Fixed. Scala should have e.g.
`data List a = Empty | Cons a (List a) deriving (Show, Read, Eq, Ord)`